src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -317,6 +317,20 @@ public override void VisitParameterDeclarationSyntax(ParameterDeclarationSyntax
                  */
                 var allowedDecoratorSyntax = syntax.Decorators.FirstOrDefault(decoratorSyntax =>
                 {
+                    if (decoratorSyntax.Expression is InstanceFunctionCallSyntax { BaseExpression: VariableAccessSyntax namespaceSyntax, Name: IdentifierSyntax nameSyntax } &&
+                        namespaceSyntax.Name.IdentifierName == "sys" &&
+                        nameSyntax.IdentifierName == "allowed")
+                    {
+                        /*
+                         * This is a workaround to unambiguously get the syntax for "@sys.allowed(...)".
+                         * We cannot get it by resolving the function symbol, since currently the binder
+                         * does not bind an InstanceFunctionCallSyntax to a symbol. This does not affect
+                         * decorator emitting though, because in the emitter we call SemanticModel.GetSymbolInfo()
+                         * which adds special handling for InstanceFunctionCallSyntax.
+                         */
```

Comment:
@majastrz do we have any plan to update the binder to bind `InstanceFunctionCallSyntax`, `PropertyAccessSyntax` and `ObjectPropertySyntax`?

/cc @miqm.

Reply:
No, It's intentionally deferred with Anthony's latest changes to support methods and property hovers.

Reply:
Got it!